### ISSUE TITLE
Gmail simultaneous logins throttled

### ISSUE THREAD

**Issue by robherring** [corecode/dma#72]
Gmail seems to have some limits on the number of simultaneous logins. It seems to be around 16. It's repeatable, so I don't think 'Temporary System Problem' below is some problem on Google's end, and sometimes the error is a max login error. Eventually everything succeeds and even though there's a 5 min retry time, I'm still seeing my machine become sluggish during this time with a bunch of processes spawned. Having some configurable throttling would be helpful.

Here's a log on one process:

dma[27748]: trying delivery
dma[27748]: using smarthost (smtp.gmail.com:587)
dma[27748]: trying remote delivery to smtp.gmail.com [173.194.78.109] pref 0
dma[27748]: SSL initialization successful
dma[27748]: using SMTP authentication for user x@gmail.com
dma[27748]: smarthost authentication: AUTH cram-md5 not available: 504 5.7.4 Unrecognized Authentication Type w72sm5765858oie.49 - gsmtp
dma[27748]: remote delivery deferred: smtp.gmail.com [173.194.78.109] failed after DATA: 421 4.3.0 Temporary System Problem.  Try again later (10). w72sm5765858oie.49 - gsmtp
dma[27748]: trying delivery
dma[27748]: using smarthost (smtp.gmail.com:587)
dma[27748]: trying remote delivery to smtp.gmail.com [173.194.78.109] pref 0
dma[27748]: SSL initialization successful
dma[27748]: using SMTP authentication for user x@gmail.com
dma[27748]: smarthost authentication: AUTH cram-md5 not available: 504 5.7.4 Unrecognized Authentication Type r10sm6650760otn.37 - gsmtp
dma[27748]: remote delivery deferred: smtp.gmail.com [173.194.78.109] failed after DATA: 421 4.3.0 Temporary System Problem.  Try again later (10). r10sm6650760otn.37 - gsmtp
dma[27748]: trying delivery
dma[27748]: using smarthost (smtp.gmail.com:587)
dma[27748]: trying remote delivery to smtp.gmail.com [173.194.78.109] pref 0
dma[27748]: SSL initialization successful
dma[27748]: using SMTP authentication for user x@gmail.com
dma[27748]: smarthost authentication: AUTH cram-md5 not available: 504 5.7.4 Unrecognized Authentication Type e6sm6633658otl.12 - gsmtp
dma[27748]: remote delivery deferred: smtp.gmail.com [173.194.78.109] failed after DATA: 421 4.3.0 Temporary System Problem.  Try again later (10). e6sm6633658otl.12 - gsmtp
dma[27748]: trying delivery
dma[27748]: using smarthost (smtp.gmail.com:587)
dma[27748]: trying remote delivery to smtp.gmail.com [173.194.78.108] pref 0
dma[27748]: SSL initialization successful
dma[27748]: using SMTP authentication for user x@gmail.com
dma[27748]: smarthost authentication: AUTH cram-md5 not available: 504 5.7.4 Unrecognized Authentication Type p184sm5756456oic.40 - gsmtp
dma[27748]: delivery successful

**Comment by corecode** [comment]
We already do exponential backoff, with a minimum of 5 minutes.  I don't see how this would make a machine sluggish.  Are you trying to deliver several mails in parallel?

**Comment by corecode** [comment]
Yes, that's how it works.  I don't think we can do anything about this.
